lib/validate.js

```diff
@@ -83,6 +83,10 @@ module.exports = {
       _.merge(entries, entry);
     } else {
       _.forEach(functions, (func, index) => {
+        const data = this.serverless.service.getFunction(func);
+        if (data.webpack === false) {
```

Comment:
Check for the runtime here - not the manually set flag, as discussed in my comment.
Additionally you create an entries array here that is shorter than the actual number of functions in the service.
This might fail in the compile and deploy steps with individual packaging when the entries have to be mapped back to the actual functions by index. A better way would be to only iterate over the supported functions (i.e. node.js) in all places then, instead of doing the check within the for loop. That would catch all possible combinations.

```
const supportedFunctions = _.filter(functions, func => !func.runtime || _.includes([ 'nodejs6.10', ... ] func.runtime);
_.forEach(supportedFunctions, (func, index) =>  {
```